packages/nextjs/app/nillion-pw-manager/page.tsx

```diff
@@ -0,0 +1,240 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import type { NextPage } from "next";
+import { useAccount } from "wagmi";
+import { CopyString } from "~~/components/nillion/CopyString";
+import Dropdown from "~~/components/nillion/Dropdown";
+import { NillionOnboarding } from "~~/components/nillion/NillionOnboarding";
+import RetrieveSecretCommand from "~~/components/nillion/RetrieveSecretCommand";
+import SecretForm from "~~/components/nillion/SecretForm";
+import { Address } from "~~/components/scaffold-eth";
+import { getUserKeyFromSnap } from "~~/utils/nillion/getUserKeyFromSnap";
+import { retrieveSecretBlob } from "~~/utils/nillion/retrieveSecretBlob";
+import { storeSecretsBlob } from "~~/utils/nillion/storeSecretsBlob";
+
+interface StoredSecrets {
+  [secretName: string]: string; // store_id
+}
+const Home: NextPage = () => {
+  const { address: connectedAddress } = useAccount();
+  const [connectedToSnap, setConnectedToSnap] = useState<boolean>(false);
+  const [userKey, setUserKey] = useState<string | null>(null);
+  const [userId, setUserId] = useState<string | null>(null);
+  const [nillion, setNillion] = useState<any>(null);
```

Comment:
Do TS types for our wasm types not work here? e.g. I imagine you should be able to use `NillionClient` rather than `any`.

Reply:
caught between a rock and a hard place here

I can't get the nillion types because without

- causing client errors: if the `import * as nillion from "@nillion/nillion-client-js-browser/nillion_client_js_browser.js";` happens without being async imported, the client complains and errors out

- creating a circular dependency: nillion js lib is imported in the client util https://github.com/NillionNetwork/scaffold-nillion/blob/main/packages/nextjs/utils/nillion/nillionClient.ts#L2

Reply:
Hmm this isn't great. Those type definitions are being generated so we should make sure they're usable and find a way to use them. I think it's okay to keep it this now until we figure that out tho. cc @wwwehr @navasvarela you may know how to make this work.

Reply:
could the types potentially be a separate import from the js client library?

Reply:
@navasvarela I noticed there's no nillion type here - just NillionClient